src/api/controllers/adminPermissionsController.js

```diff
@@ -0,0 +1,192 @@
+import models from '../models';
+import status from '../../helpers/constants/status.codes';
+import paginateUser from '../../helpers/generate.pagination.details';
+
+const { User } = models;
+/**
+ * containing all user's model controllers (signup, login)
+ *
+ * @export
+ * @class Auth
+ */
+export default class adminPermissions {
+  /**
+   * This function gets all the users that are stored in the User table
+   *
+   * @static
+   * @param {Object} req the request
+   * @param {Object} res the response to be sent
+   * @memberof users
+   * @returns {Object} res
+   */
+  static async getUsers(req, res) {
+    // default constants
+    const defaultLimit = 10;
+    const defaultOffset = 0;
+    const result = {};
+    const emptyUserArray = 0;
+
+    const limit = req.query.limit || defaultLimit;
+    const offset = req.query.offset || defaultOffset;
+    const users = await User.findAndCountAll({
+      offset,
+      limit
+    });
+
+    result.status = status.OK;
+    result.pagedUsers = paginateUser(users.count, users.rows, offset, limit);
+    result.users = users.rows;
+    if (users.rows.length > emptyUserArray) {
+      res.status(status.OK).json({
+        result
```

Comment:
@Ramadhan0 there was feedback about returning the status in addition with the result here.

Reply:
i resolved this